// my-app/src/components/PopularCities/cityData.js
// cityData.js
import Lahore from '../../assets/images/Lahore2.jpg'
import Karachi from '../../assets/images/Karachi1.jpeg';
import Islamabad from '../../assets/images/Islamabad1.jpg';
import Queta from '../../assets/images/Balochistan.jpg';
import Mansehra from '../../assets/images/Mansehra.jpg';


export const citiesData = {
    Lahore: {
      name: "Lahore",
      description:"Lahore, the cultural heart of Pakistan, is known for its rich history, stunning Mughal architecture, and vibrant street life. As a city of art, education, and cuisine, it offers a unique blend of tradition and modernity.",
      image: Lahore,
      transportOptions: ["Public Transport", "Private Transport", "Business Tours", "Local Visit"],
    },
    
    Karachi: {
      name: "Karachi",
      description: "Karachi, Pakistan's largest city and economic hub, is a bustling metropolis known for its diverse culture and coastline. It serves as the country's center for commerce, industry, and a vibrant urban lifestyle.",
      image:Karachi,
      transportOptions: ["Nature & Adventure", "Private Transport", "Local Visit"],
    },
    Islamabad: {
      name: "Ialamabad",
      description: "Islamabad, the capital city of Pakistan, is renowned for its modern infrastructure, lush greenery, and serene environment. Nestled at the foothills of the Margalla Hills, it offers a peaceful contrast to the country's bustling urban centers.",
      image: Islamabad,
      transportOptions: ["Public Transportations", "Nature & Adventure", "Private Transport", "Business Tours", "Local Visit"],
    },
    Queta: {
      name: "Queta",
      description: "Balochistan, Pakistan's largest province by area, is rich in natural resources and offers stunning landscapes, from arid deserts to rugged mountains. Its unique culture and traditions reflect the resilience and hospitality of its people.",
      image: Queta,
      transportOptions: ["Public Transportations", "Nature & Adventure", "Private Transport", "Business Tours", "Local Visit"],
    },
    Mansehra: {
      name: "Mansehra",
      description: "Mansehra, a picturesque city in northern Pakistan, is known for its lush green valleys and scenic beauty. It serves as a gateway to popular tourist destinations like Kaghan Valley and Naran, offering a peaceful retreat for nature lovers",
      image: Mansehra,
      transportOptions: ["Public Transportations", "Nature & Adventure", "Private Transport", "Business Tours", "Local Visit"],
    },
    Pashawar: {
      name: "Pashawar",
      description: "Mansehra, a picturesque city in northern Pakistan, is known for its lush green valleys and scenic beauty. It serves as a gateway to popular tourist destinations like Kaghan Valley and Naran, offering a peaceful retreat for nature lovers",
      image: Mansehra,
      transportOptions: ["Public Transportations", "Nature & Adventure", "Private Transport", "Business Tours", "Local Visit"],
    },
    
    // Add other cities similarly
  };
